base/strings/annotated.jl

```diff
@@ -386,3 +386,62 @@ annotations(s::SubString{<:AnnotatedString}, pos::UnitRange{<:Integer}) =
 Get all annotations of `chr`.
 """
 annotations(c::AnnotatedChar) = c.annotations
+
+## AnnotatedIOBuffer
+
+struct AnnotatedIOBuffer <: IO
+    io::IOBuffer
+    annotations::Vector{Tuple{UnitRange{Int}, Pair{Symbol, Any}}}
+    dict::ImmutableDict{Symbol, Any}
```

Comment:
I suspect that this `dict` field belongs in `IOContext`, not the buffer itself. (e.g. `IOContext{AnnotatedIOBuffer}`).

```julia
julia> dump(IOContext(IOBuffer(), :a => 2))
IOContext{IOBuffer}
  io: IOBuffer
    data: Array{UInt8}((32,)) UInt8[0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00  …  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00]
    reinit: Bool false
    readable: Bool true
    writable: Bool true
    seekable: Bool true
    append: Bool false
    size: Int64 0
    maxsize: Int64 9223372036854775807
    ptr: Int64 1
    mark: Int64 -1
  dict: Base.ImmutableDict{Symbol, Any}
    parent: Base.ImmutableDict{Symbol, Any}
      parent: #undef
      key: #undef
      value: #undef
    key: Symbol a
    value: Int64 2
```

Reply:
Ah yes, you can just wrap an `AnnotatedIOBuffer` in an IO context. At the time I added this, I was thinking that since this is useful for printing things, it could be nice if an `AnnotatedIOBuffer` could also contain properties like `displaysize`.

Upon reflection, I'm not sure whether this is actually a good idea, but it does seem like something that could be added later if it turns out that we'd like it: So I'm thinking remove the dict and we can cross that bridge later if we want.